src/language-server/project/rover/DocumentSynchronization.ts

```diff
@@ -229,4 +251,48 @@ export class DocumentSynchronization {
       position: match.position,
     });
   }
+
+  handlePartDiagnostics(params: PublishDiagnosticsParams) {
```

Comment:
Diagnostics from the upstream LSP are for "virtual files" like `${realFileName}/${index}` - and we have to map these to the correct document part, replace the diagnostics of that part with the reported diagnostics, and then report the diagnostics for the full file (accumulation of virtual files) back to VSCode.